### ISSUE TITLE
Supplier sourced items in Sub-contracting module

### ISSUE THREAD

**Issue by KanchanChauhan** [frappe/erpnext#23475]
**Problem**

- In a workflow where a Supplier or Sub-Contractor assembles or produces the product, some of the raw material is provided by the user, and some are sourced by the Supplier himself.
- For the user, such items do not have any stock value or valuation rate.
- They should exist only in BOM and Purchase Order.
- They should also be visible in Purchase Order to Sub-Contractor so as to know that they have to source it themselves.
- Currently, there is no way to flag such items.

**Solution**

- Create a child table in BOM for Supplier Sourced Items.
- Also show these items in the Purchase Order at zero value in the separate child table of Sub-Contractor

**Comment by asbasawaraj** [comment]
@KanchanChauhan The items sourced by the sub-contractor should be added in the BOM and they should be shown in PO so that sub-contractor knows he needs to sources those items on his own. This part I have clarity.

Work order doesn't come into the picture in the above situation. How does adding those details benefit in the work order help?

**Comment by KanchanChauhan** [comment]
@asbasawaraj I understand, there should not be a work order for these kinds of BOMs. Just trying to cover up all the scenarios, it will just be a read-only table pulled from BOM if someone decides to have a Supplier Work order also for their reference.
Also to let you know, we have started the development and it will be ready soon.

**Comment by asbasawaraj** [comment]
@KanchanChauhan Today, I, @rohitwaghchaure and @marination had a discussion on this. While we have to support more use cases, we should not also try to keep the product simple to use. It's hard to strike that balance.

We are trying to come up with alternative designs as well. 

Sure, It will be good to have PR.

**Comment by creamdory** [comment]
Hi @asbasawaraj about this. As what @KanchanChauhan wrote, we suggest to have it in another child table. This table will be visible also in the PO for subcontracting. We decided to do this instead of the tag in the item (since an item is not always sourced by supplier so this is just BOM level) in order to exclude these item in the transfer for subcontracting. 

> @KanchanChauhan The items sourced by the sub-contractor should be added in the BOM and they should be shown in PO so that sub-contractor knows he needs to sources those items on his own. This part I have clarity.
> 
> Work order doesn't come into the picture in the above situation. How does adding those details benefit in the work order help?

**Comment by rmehta** [comment]
Why not just attach the BOM with the Purchase Order? This way we don't need to copy the items in the PO? Instead of another child table, these can be just tagged in the BOM "Supplier bought", right?

**Comment by KanchanChauhan** [comment]
@rmehta In PO we already show the items from BOM which we will be providing them in separate table as Supplied Items so kind of made sense to have another table which will show items they have to procure themselves as Supplier Sourced Items. Also, the quantity is a factor, in PO we will show raw material qty based on Item qty, in BOM it is just for 1.

**Comment by rmehta** [comment]
@KanchanChauhan yes, the qty multiplier is a point.

I still believe this additional functionality should not be part of PO but an additional report / print format that can be attached with the PO while sending it to the supplier.

**Comment by marination** [comment]
@creamdory @KanchanChauhan We did give this more thought , and decided not to proceed with it. The main pain points being:
- The table will just add more bulk. More importantly it is also very confusing to someone not familiar with this workflow (which will be perhaps 80% of our manufacturing user base), as it is a very specific case. 
- We do not know of many users that actually require this at the moment, no strong reason to add it to the core.

To summarise it, this is something that can be driven by customizations and we don't see any way it does not lower the UX.

**Comment by KanchanChauhan** [comment]
@marination @rmehta @asbasawaraj Thanks for your input on this. We will proceed with the customization, that will be fairly simple.

But it is a very common scenario in manufacturing and sub-contracting where Suppliers (sub-contractor) need to source some raw material themselves for example nuts and bolts. These raw materials need to be part of BOM but have no stock entry or accounting impact.

If we add them to BOM Item, it has issues with material transfer and consumption entries, even if you flag them, you have to have conditions at lots of places while making stock entries or Purchase Order. This approach seemed less invasive.

Since we are the manufacturing company ourselves and another company also requested a similar feature, we thought it will be best if it was in the core.

**Comment by rmehta** [comment]
This is bad design because the data is redundant on the PO and is unconnected to the core functionality of the PO.

In either case, you will have to add them as BOM Items if you want to make the process repeatable. Also there is a supplied items table in PO, maybe these items can be added to that table with a flag that they have to be bought by the supplier.

Every new table leads to exponential complexity and long term maintenance will fall on the core maintainer group. Flagging the items is the least invasive way, and seems like a win-win approach.

**Comment by KanchanChauhan** [comment]
@rmehta I do understand, so you are ok with the idea of having a flag in BOM Item in BOM and Supplied Item in PO. And we can exclude these items from consumption stock entries.

**Comment by rmehta** [comment]
Yep, seems like a good design. Minimum changes in structure.
